p2p/src/authenticated/lookup/network.rs

```diff
@@ -119,7 +119,11 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng + RNetwork + Metr
     ///
     /// After the network is started, it is not possible to add more channels.
     pub fn start(mut self) -> Handle<()> {
-        self.context.spawn_ref()(self.run())
+        let context = self.context.take();
+        context.spawn(move |context| async move {
```

Comment:
While it seems like it would be possible to automatically capture the `context` on the other side of this function, you'd be shocked.

We need to expose context fully because the "Builder API" consumes self.

Reply:
I tried very hard to get this down to:
```
self.context.spawn(self.run())
```

But couldn't do it (as the self.context would partially consume self).